[AlphaFilters] support sectioned filters (#9197)
Shopify/polaris: .changeset/loud-dancers-live.md

---
'@shopify/polaris': minor
---

> new prop additionalFilterSections in IndexFilters and AlphaFilters

Reviewer: Added the `additionalFilterSections` prop to `IndexFilters` and `AlphaFilters` to support sectioning the additional filters list

> Added the `filterSections` prop to `IndexFilters` and `AlphaFilters` to support sectioning the additional filters list

Reviewer: ```suggestion
Added support for sectioning the `filters` of `IndexFilters` and `AlphaFilters` 
```
